src/crystal/system/unix/group.cr

```diff
@@ -0,0 +1,58 @@
+require "c/grp"
+
+module Crystal::System::Group
+  private GETGR_R_SIZE_MAX = 1024 * 16
+
+  private def extract_members(gr_mem)
+    members = Array(String).new
```

Comment:
(optional!) One way to optimize this is to first count `n`. Then we can do `members = Array(String).new(n)` which should avoid potential memory reallocations.